### DISCUSSION TITLE
(API) Can extension store data?

### DISCUSSION THREAD

**Discussion by piw-piw** [ScratchAddons/ScratchAddons#6527] [Help]
Can extension store data, like arrays? Not as setting, but hidden variable.

**Comment by rgantzos** [comment]
Chrome has a storage API that the extension uses in various ways to store data. It uses it to store settings and more.

**Reply by CST1229** [reply]
I assume they mean an *addon*, not the extension itself.

**Comment by WorldLanguages** [comment]
Define "store".

**Reply by WorldLanguages** [reply]
I guess you mean storing permanently, as if it was a setting.

Also, could you tell us what you're working on, instead of doing random specific questions? I think we could be more helpful that way.

**Reply by piw-piw** [reply]
I'm experimenting with saving project data, like game saves etc., so you don't have to copy & paste your **save code**.
It could be used by custom blocks to save and load string assigned to project ID.

**Reply by Samq64** [reply]
There's already a TurboWarp extension for that. New blocks can't be added to Scratch Addons because they need to be compatible with Scratch. The debugger blocks are an exception because they're just a pink block that doesn't do anything in normal Scratch and it's only meant for project development.

**Reply by piw-piw** [reply]
Save & load blocks also should do nothing in scratch.

**Reply by WorldLanguages** [reply]
Well, we would need to discuss this in an issue. If you're doing this as a personal experiment, you can use `localStorage`, but keep in mind we might not accept your submission.
